lib/classes/markus_csv.rb

```diff
@@ -15,14 +15,16 @@ def self.generate(objects, headers=[], &gen_csv)
   end
 
   # Performs an action for each object in a collection represented by
-  # a CSV string. 'input' is the input string, and parse_obj is a block which
-  # takes a line and performs an action, or raises CSVInvalidLineError.
+  # a CSV string. 'input' is the input string, 'header_count' is the
+  # number of header lines to remove from the displayed valid_line_count
+  # and parse_obj is a block which takes a line and performs an action,
+  # or raises CSVInvalidLineError.
   # Returns a result hash, containing a success message with the number of
   # successful rows parsed, as well as an error message, consisting of one
   # of the following:
   #   1) A string listing all erroneous lines.
   #   2) A more generic error message for invalid files.
-  def self.parse(input, options = {}, &parse_obj)
+  def self.parse(input, options = {}, header_count = 0, &parse_obj)
```

Comment:
Oh geez, it looks like it's the `CSV.parse` method specifically that is rejecting options that aren't recognized. I forgot that we passed the options hash to this method. In this case, use `options.delete(:header_count)` to both extract the option and remove it from `options`. (This returns `nil` if the key is not present, so you still need to specify the default value.)